Backmerge Commcare 2.58

avazirna opened the pull request:
## Product Description
This PR back-merges changes from CommCare 2.58.0 into the master branch.

cross-request: https://github.com/dimagi/commcare-core/pull/1493

avazirna commented:
@damagatchi retest this please

avazirna commented:
@coderabbitai summary

shubham1g5 commented on app/AndroidManifest.xml:
```diff
@@ -2,7 +2,7 @@
           xmlns:tools="http://schemas.android.com/tools"
           android:versionCode="106"
-          android:versionName="2.59">
+          android:versionName="2.58.0">
```
version name in `master` should be `2.59`

shubham1g5 commented on app/build.gradle:
```diff
@@ -48,8 +48,9 @@ dependencies {
     testImplementation 'org.jetbrains.kotlinx:kotlinx-coroutines-test:1.7.3'
     testImplementation 'io.mockk:mockk:1.12.7'
-    testImplementation 'org.json:json:20250517'
+    testImplementation 'org.json:json:20231013'
```
should not be downgrading

shubham1g5 commented on app/build.gradle:
```diff
@@ -106,6 +107,7 @@ dependencies {
     implementation 'com.journeyapps:zxing-android-embedded:3.6.0'
     implementation 'com.google.firebase:firebase-analytics:20.1.2'
     implementation 'com.google.firebase:firebase-messaging:21.1.0'
+    implementation 'com.google.firebase:firebase-crashlytics:17.2.1'
```
duplicate

shubham1g5 commented on app/src/org/commcare/CommCareApplication.java:
```diff
@@ -219,9 +218,6 @@ public void onCreate() {
-        if (!BuildConfig.DEBUG) {
-            FirebasePerformance.getInstance().setPerformanceCollectionEnabled(true);
-        }
 
```
mistake ?

shubham1g5 commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-bin.zip
-zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
should not downgrade.

shubham1g5 commented on build.gradle:
```diff
@@ -1,13 +1,13 @@
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:8.6.1'
+        classpath 'com.android.tools.build:gradle:8.1.1'
```
should not downgrade.

shubham1g5 reviewed (changes requested):
Think bunch of errors related to how conflicts were resolved in this branch.

avazirna commented:
> Think bunch of errors related to how conflicts were resolved in this branch.

Yes, I had to re-run the `deploy release` step due to an error we found after the first run. Maybe there is a better way to do this, I deleted the release tags and reverted a couple of commits.

avazirna commented:
@shubham1g5 I've reverted the merge commit revert - [7dce414](https://github.com/dimagi/commcare-android/pull/3272/commits/7dce41403faf43c378d00632fc5f1540dc9d8578)
